File: front/medicine/src/app/shared/models/entity/Drug.ts
import { User } from "./User";
import { TranslatedDrugs } from "./translate/TranslatedDrugs";
import { ActiveElement } from "./ActiveElement";
import { SimilarDrugs } from "./SimilarDrugs";
import { DrugCategory } from "./DrugCategory";


export class Drug {
  title?: string;
  oneUnitSizeInGramm!: number;
  drugCategories: DrugCategory[] = [];
  similarDrugs!: SimilarDrugs;
  similarDrugsId!: number;
  activeElements: ActiveElement[] = [];
  translations: TranslatedDrugs[] = [];
  user?: User;
  userId?: number;
  id!: number;
  createdAt?: Date;
}

File: src/Entities/Medicine.Entities/Models/Drugs/Drug.cs
using Medicine.Entities.Models.Base.Interfaces;
using Medicine.Entities.Models.Translated;

namespace Medicine.Entities.Models.Drugs
{
    /// <summary>
    /// Medicine,drugs
    /// </summary>
    public class Drug : TranslationEntity<TranslatedDrugs>
    {
        public string? Title { get; set; }
        public double OneUnitSizeInGramm { get; set; }

        public List<DrugCategory> DrugCategories { get; set; } = new();

        public SimilarDrugs SimilarDrugs { get; set; } = new();
        public int SimilarDrugsId { get; set; }

        public List<ActiveElement> ActiveElements { get; set; } = new();

    }
}
